testsuites/benchmark/src/main/java/org/eclipse/rdf4j/benchmark/rio/util/DataSetGenerator.java

```diff
@@ -0,0 +1,420 @@
+	private static final String WORD_DIR = "generated-data/words.txt";
+	private static final int TOTAL_PREDS = 500;
+	private static final int AVG_GRAPH_SIZE = 500;
+	private final ValueFactory factory;
+	private final StringBuilder valueBuilder;
+	private final AtomicInteger contextCount = new AtomicInteger(0);
+	private final AtomicInteger maxCurrentContextCount = new AtomicInteger(0);
+	private ArrayList<String> dictionary;
+	private ArrayList<IRI> predicateIRIs;
+	private ArrayList<IRI> contextIRIs;
+	private Random randomResource;
+	private Random randomProbability;
+	private BNode prevBNode;
+	private IRI currentContext;
+	private int totalWords;
+	private int avgWordSize;
+
+	public DataSetGenerator() {
+		this.factory = SimpleValueFactory.getInstance();
+		this.valueBuilder = new StringBuilder();
+		this.randomResource = new Random(64);
+		this.randomProbability = new Random(64);
+		this.avgWordSize = 0;
+		fetchWords();
+		generatePredicateList();
+	}
+
+	public File generateStatementsFile(RDFFormat format, String fileName, int percentBNodes, int percentLiterals,
+			int textMinLength, int textMaxLength, int totalStatements, boolean isTextOnly,
+			boolean withContext) throws IOException {
+		assert percentBNodes + percentLiterals <= 100 && totalStatements > 0 && textMaxLength > 0
+				&& textMinLength < textMaxLength;
+		File file = createFile(format, fileName);
+		if (format.equals(RDFFormat.BINARY)) {
+			FileOutputStream out = new FileOutputStream(file);
+			if (withContext) {
+				generateContextList(totalStatements);
+				writeBinaryStatementsWithContext(out, percentBNodes, percentLiterals, textMinLength, textMaxLength,
+						totalStatements, isTextOnly);
+				maxCurrentContextCount.set(0);
+				contextCount.set(0);
+				currentContext = null;
+			} else {
+				writeBinaryStatementsWithoutContext(out, percentBNodes, percentLiterals, textMinLength,
+						textMaxLength,
+						totalStatements, isTextOnly);
+			}
+		} else {
+			BufferedWriter out = new BufferedWriter(
+					new OutputStreamWriter(new FileOutputStream(file), StandardCharsets.UTF_8));
+			if (withContext) {
+				generateContextList(totalStatements);
+				writeStatementsWithContext(out, format, percentBNodes, percentLiterals, textMinLength, textMaxLength,
+						totalStatements, isTextOnly);
+				maxCurrentContextCount.set(0);
+				contextCount.set(0);
+				currentContext = null;
+			} else {
+				writeStatementsWithoutContext(out, format, percentBNodes, percentLiterals, textMinLength,
+						textMaxLength,
+						totalStatements, isTextOnly);
+			}
+		}
+		prevBNode = null;
+		randomResource = new Random(64);
+		randomProbability = new Random(64);
+		return file;
+	}
+
+	private File createFile(RDFFormat format, String fileName) throws IOException {
+		File file = new File(Strings.append(File.separator).to(System.getProperty("java.io.tmpdir")) + fileName + "."
+				+ format.getDefaultFileExtension());
+		assert file.createNewFile();
+		return file;
+	}
```

Comment:
Just one last thing @tokovach . Could you create a temp file with a UUID instead of the filename. Since you are passing the file back up the stack all the way to the jmh benchmark code, we don't really need to care what the file is called (eg. you can remove the fileName argument). By using a UUID we will also remove any chance of clashing with an existing file. This dataset generator is very useful and we might end up using it other places, so would be great if it was resilient :)